backend/api/serializers.py

```diff
@@ -249,9 +249,21 @@ def update(self, instance, validated_data):
         instance.last_name = validated_data.get('last_name', instance.last_name)
         instance.bio = validated_data.get('bio', instance.bio)
         instance.city = validated_data.get('city', instance.city)
+        instance.is_corporate_user = validated_data.get('is_corporate_user', instance.is_corporate_user)
         corporate_profile = validated_data.get('corporate_profile', instance.corporate_profile)
-        corp = CorporateUserProfile(url=corporate_profile.get('url'))
-        instance.corporate_profile = corp
+        if not instance.is_corporate_user:
+            # case: corporate user is disabled
+            instance.corporate_profile = None
+        else:
+            if instance.corporate_profile:
+                # case: corporate user profile is updated
+                instance.corporate_profile.url = corporate_profile.get('url', None)
```

Comment:
```suggestion
                instance.corporate_profile.url = corporate_profile.get('url', None)
                instance.corporate_profile.save()
```

`instance.corporate_profile.save()` can be moved here and removed from line#265.